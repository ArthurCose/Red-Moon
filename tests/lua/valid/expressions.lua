print("values:")
print "hello"
print("hi", 'hello', [[world]], true, 1.0, 2, false, 0xFF, nil)

print("\nshort circuit:")
print(false or 1, 2 or false, true and 3, false and 4, true and false)

print("\nunary:")
print(not false and 3)

print("\narithmetic:")
print(1 + 2 * 2, 3 * 2 + 1, (1 + 2) * 2, 1. + 2, 1. << 2, 2 * -3, 2 ^ 2 ^ -1)

print("\nboolean comparison:")
print(1. == 1, 1. ~= 1, 1 < 2, 1 > 2, 1 >= 2, 1 <= 2, 1 >= 1, 1 <= 1)

print("\nstring comparison:")
print("a" < "b", "a" > "b", "a" == "b", "a" == "a", "a" >= "b", "a" >= "a")

print("\nstring arithmetic:")
print("1" + "1", "1" + "1.0", "1" + 1, "1" + 1.0, "1.0" + 1)

print("\nfunction comparison:")
local foo = function() end
local bar = function() end
print(foo == foo, foo ~= foo, foo == bar)

print("\nconcat:")
print("a" .. "b")
print("a" .. 1)
print((1) .. "b")

print("\nlength:")
print(#"1234")
print(#{ 1, 2, 3, 4, 5 })
