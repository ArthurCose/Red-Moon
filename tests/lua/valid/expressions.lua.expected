values:
hello
hi	hello	world	true	1.0	2	false	255	nil

short circuit:
1	2	3	false	false

unary:
3

arithmetic:
5	7	6	3.0	4	-6	1.4142135623730951

boolean comparison:
true	false	true	false	false	true	true	true

string comparison:
true	false	false	true	false	true

string arithmetic:
2	2.0	2	2.0	2.0

function comparison:
true	false	false

concat:
ab
a1
1b

length:
4
5
